x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportGetDataFrameAnalyticsStatsAction.java

```diff
@@ -137,9 +137,10 @@ protected void doExecute(Task task, GetDataFrameAnalyticsStatsAction.Request req
             response -> {
                 List<String> expandedIds = response.getResources().results().stream().map(DataFrameAnalyticsConfig::getId)
                     .collect(Collectors.toList());
+                request.setCount(response.getResources().count());
                 request.setExpandedIds(expandedIds);
                 ActionListener<GetDataFrameAnalyticsStatsAction.Response> runningTasksStatsListener = ActionListener.wrap(
-                    runningTasksStatsResponse -> gatherStatsForStoppedTasks(request.getExpandedIds(), runningTasksStatsResponse, listener),
+                    runningTasksStatsResponse -> gatherStatsForStoppedTasks(request, runningTasksStatsResponse, listener),
```

Comment:
Rather than adding a `count` field to the request -which feels odd because it is part of the response not a request param-  you could use `response.getResources().count()` in the `gatherStatsForStoppedTasks` listener

```suggestion
                    runningTasksStatsResponse -> gatherStatsForStoppedTasks(request, runningTasksStatsResponse, ActionListener.wrap(finalResponse -> { listener.onResponse(new Response(finalResponse.getStats(), response.getResources().count() ),
                    ...
```

Reply:
Ah, that is a very good point. I'll do that.